app.js

```diff
@@ -1,29 +1,32 @@
 var express=require('express');
 var expressLayouts = require('express-ejs-layouts');
-var session=require('express-session');
-var favicon=require('serve-favicon');
 var app=express();
+app.set('view engine','ejs');
+app.use(expressLayouts);
+app.use(express.static('./public'));
+
 // dotenv
 var dotenv = require('dotenv');
 dotenv.config();
-//mongodb connection
-var mongoose = require('mongoose');
-mongoose.connect('mongodb+srv://' + process.env.MONGOUSER + ':' + process.env.MONGOPASSWORD + '@cluster0.jlrkv.mongodb.net/'+process.env.MONGODBNAME+'?retryWrites=true&w=majority', { useNewUrlParser: true })
+
+//passport
+var passport = require('passport');
+app.use(passport.initialize());
+app.use(passport.session());
+passport = require('./config/passport');

```

Comment:
you're overwriting the passport var from line 13 with this require. Did you mean to require the passport module from `./config/passport` and use that in your app ? If so just do 
```javascript
const passport = require('./config/passport')
app.use(passport.initialize());
app.use(passport.session());
```

Reply:
For some reason I thought that I first need to run initialize and session methods from the passport module, and then overwrite it. I've corrected it.

Reply:
As a clarification, this correction did not solve the login problem. I've omitted to mention this, sorry

Reply:
please push changes so we can take another look :)